draft-ietf-httpbis-message-signatures.md

```diff
@@ -382,7 +343,153 @@ This example shows a canonicalized value for the parameters of a given signature
   created=1618884475;expires=1618884775
 ~~~
 
-Note that an HTTP message could contain multiple signatures, but only the signature parameters used for the current signature are included in this field.
+Note that an HTTP message could contain multiple signatures, but only the signature parameters used for the current signature are included in this entry.
+
+### Method {#content-request-method}
+
+The `@method` identifier refers to the HTTP method of a request message. The value of is canonicalized by taking the value of the method as a string. Note that the method name is case-sensitive as per {{!SEMANTICS=RFC7231}}, and conventionally standardized method names are uppercase US-ASCII.
+
+For example, the following request message:
+
+~~~
+POST /path?param=value HTTP/1.1
+Host: www.example.com
+~~~
+
+Would result in the following `@method` value:
+
+~~~
+"@method": POST
+~~~
+
+### Authority {#content-request-authority}
+
+The `@authority` identifier refers to the authority component of the target of the HTTP request message. In HTTP 1.1, this is usually conveyed using the `Host` header, while in HTTP 2 and HTTP 3 it is conveyed using the `:authority` pseudo-header. The value is the fully-qualified authority component of the request, comprised of the host and, optionally, port of the request target, as a string.
+
+For example, the following request message:
+
+~~~
+POST /path?param=value HTTP/1.1
+Host: www.example.com
+~~~
+
+Would result in the following `@authority` value:
+
+~~~
+"@authority": www.example.com
+~~~
+
+### Scheme {#content-request-scheme}
+
+The `@scheme` identifier refers to the scheme of the target URL of the HTTP request message. The value is the scheme as a string.
+
+For example, the following request message requested over plain HTTP:
+
+~~~
+POST /path?param=value HTTP/1.1
+Host: www.example.com
+~~~
+
+Would result in the following `@scheme` value:
+
+~~~
+"@scheme": http
+~~~
+
+### Request Origin {#content-request-origin}
+
+The `@request-origin` identifier refers to the full origin component of the HTTP request message. The value is combination of the absolute path and query components of the request URL.
+
+For example, the following request message:
+
+~~~
+POST /path?param=value HTTP/1.1
+Host: www.example.com
+~~~
+
+Would result in the following `@request-origin` value:
+
+~~~
+"@request-origin": /path?param=value
+~~~
+
+The following OPTIONS request message:
+
+~~~
+OPTIONS * HTTP/1.1
+Host: www.example.com
+~~~
+
+Would result in the following `@request-origin` value:
+
+~~~
+"@request-origin": *
```

Comment:
Would it?

Also, it might be good to talk about what distinguishes `@request-origin` from `@authority`.

Reply:
I wasn't sure about this -- it's supposed to be the path+query portion of the request, the bit after the verb on the HTTP/1.1 request line.

Reply:
Ah. 'origin' is the (scheme, host, port) tuple. If you want path+query, we should invent another term.

Reply:
Changed the name back to request-target but with largely the same value definition as here.